statsmodels/genmod/generalized_linear_model.py

```diff
@@ -784,18 +790,50 @@ def estimate_tweedie_power(self, mu, method='brentq', low=1.01, high=5.):
         power : float
             The estimated shape or power
         """
+        FLOAT_EPS = np.finfo(float).eps
+        mu = np.clip(mu, FLOAT_EPS, np.inf)
+        weights = self.iweights
+
         if method == 'brentq':
-            from scipy.optimize import brentq
-
-            def psi_p(power, mu):
-                scale = ((self.iweights * (self.endog - mu) ** 2 /
-                          (mu ** power)).sum() / self.df_resid)
-                return (np.sum(self.iweights * ((self.endog - mu) ** 2 /
-                               (scale * (mu ** power)) - 1) *
-                               np.log(mu)) / self.freq_weights.sum())
-            power = brentq(psi_p, low, high, args=(mu))
+            from scipy.optimize import brentq, minimize_scalar
+
+            low = kwargs.get('low', 1.01)
+            high = kwargs.get('high', 1.99)
+
+            def difference(p):
+                scale = ((self.freq_weights * (self.endog - mu) ** 2 /
+                          mu ** p).sum() / self.df_resid)
+                scale /= self.var_weights
+
+                def psi_p(power):
+                    psi = ((self.endog - mu) ** 2 / (scale * mu ** power) - 1)
```

Comment:
var_weights always needs to go in in the same way as inverse scale (1 / scale), i.e. line 806 above
so, this looks fine to me.